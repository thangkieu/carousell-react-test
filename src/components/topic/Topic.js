import React from 'react';
const Topic = ({ topic }) => (
  <div>
    <h4>{topic.title}</h4>
    <p>{topic.content}</p>
    <button>Vote</button>
  </div>
);

export default Topic;
