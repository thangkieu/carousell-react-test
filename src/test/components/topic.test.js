import React from 'react';
import Topic from '../../components/topic';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

it('Should match snapshot with full data', () => {
  const component = renderer.create(<Topic topic={{
    title: 'Topic 1',
    content: 'Topic 1 content',
    upvote: 10,
    downvote: 5
}} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Functions pass to component should be called', () => {
  const onUpVoteTopic =jest.fn();
  const onDownVoteTopic =jest.fn();

  const component = mount(<Topic topic={{
    title: 'Topic title',
    content: 'Topic content'
  }} onUpVoteTopic={onUpVoteTopic} onDownVoteTopic={onDownVoteTopic} />);

  component.find('.btn-downvote').simulate('click');
  expect(onDownVoteTopic).toBeCalled();

  component.find('.btn-vote').simulate('click');
  expect(onUpVoteTopic).toBeCalled();
})
