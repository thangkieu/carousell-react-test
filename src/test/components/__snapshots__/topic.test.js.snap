// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot with full data 1`] = `
<div
  className="topic-item"
>
  <h4>
    Topic 1
  </h4>
  <p>
    Topic 1 content
  </p>
  <button
    className="btn btn-default btn-sm btn-vote"
    onClick={undefined}
  >
    <span>
      10
       | 
    </span>
    <span>
      Vote
    </span>
  </button>
  <button
    className="btn btn-link btn-sm btn-downvote"
    onClick={undefined}
  >
    <span>
      5
       | 
    </span>
    <span>
      DownVote
    </span>
  </button>
</div>
`;
